api/v1alpha1/bgppeer_webhook.go

```diff
@@ -0,0 +1,182 @@
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging bgppeer-webhook
+var bgppeerlog = logf.Log.WithName("bgppeer-webhook")
+
+func (bgpPeer *BGPPeer) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(bgpPeer).
+		Complete()
+}
+
+const (
+	BGPDefaultPort     = 179
+	BGPDefaultHoldTime = 90 * time.Second
+)
+
+var _ webhook.Defaulter = &BGPPeer{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (bgpPeer *BGPPeer) Default() {
+	bgppeerlog.Info("default", "name", bgpPeer.Name)
+
+	if bgpPeer.Spec.Port == 0 {
+		bgpPeer.Spec.Port = BGPDefaultPort
+	}
+	if bgpPeer.Spec.HoldTime == 0 {
+		bgpPeer.Spec.HoldTime = BGPDefaultHoldTime
+	}
+}
+
+//+kubebuilder:webhook:verbs=create;update,path=/validate-metallb-io-v1alpha1-bgppeer,mutating=false,failurePolicy=fail,groups=metallb.io,resources=bgppeers,versions=v1alpha1,name=bgppeervalidationwebhook.metallb.io,sideEffects=None,admissionReviewVersions=v1
+
+var _ webhook.Validator = &BGPPeer{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (bgpPeer *BGPPeer) ValidateCreate() error {
+	bgppeerlog.Info("validate create", "name", bgpPeer.Name)
+	existingBGPPeersList, err := getExistingBGPPeers()
+	if err != nil {
+		return err
+	}
+	return bgpPeer.validateBGPPeer(existingBGPPeersList)
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (bgpPeer *BGPPeer) ValidateUpdate(old runtime.Object) error {
+	bgppeerlog.Info("validate update", "name", bgpPeer.Name)
+	existingBGPPeersList, err := getExistingBGPPeers()
+	if err != nil {
+		return err
+	}
+	return bgpPeer.validateBGPPeer(existingBGPPeersList)
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (bgpPeer *BGPPeer) ValidateDelete() error {
+	bgppeerlog.Info("validate delete", "name", bgpPeer.Name)
+
+	return nil
+}
+
+func (bgpPeer *BGPPeer) validateBGPPeer(existingBGPPeersList *BGPPeerList) error {
+	var allErrs field.ErrorList
+	if err := bgpPeer.validateBGPPeersMyASN(existingBGPPeersList); err != nil {
+		allErrs = append(allErrs, err)
+	}
+	if err := bgpPeer.validateBGPPeersRouterID(existingBGPPeersList); err != nil {
+		allErrs = append(allErrs, err)
+	}
+	if err := bgpPeer.validateBGPPeersPort(existingBGPPeersList); err != nil {
+		allErrs = append(allErrs, err)
+	}
+	if err := bgpPeer.validateBGPPeersHoldTime(existingBGPPeersList); err != nil {
+		allErrs = append(allErrs, err)
+	}
+	if len(allErrs) == 0 {
+		return nil
+	}
+
+	err := apierrors.NewInvalid(
+		schema.GroupKind{Group: "metallb.io", Kind: "BGPPeer"},
+		bgpPeer.Name, allErrs)
+	return err
+}
+
+func (bgpPeer *BGPPeer) validateBGPPeersMyASN(existingBGPPeersList *BGPPeerList) *field.Error {
+	myASN := bgpPeer.Spec.MyASN
+
+	for _, BGPPeer := range existingBGPPeersList.Items {
+		if myASN != BGPPeer.Spec.MyASN {
```

Comment:
what is this check about? The way I read it, all the peers must have the same myASN ?

Reply:
@cgoncalves I get an email from u about ur review but I don't see it here, I just wanted to be clear here myASN hs nothing to do eith eBGP vs iBGP that AS used when u define the peer is what will be mapped to remote-AS for FRR MyASN just represent the BGP speaker instance and all sessions in this instance should have one myASN I originally suggested having global BGP config and stick all this there instead of replicating it for every peer but that suggestion wasn't approved

Reply:
This is the same discussion as https://github.com/metallb/metallb/pull/958#discussion_r711333462

Meaning, can we present ourselves with different myASNs to different peers?
The current api seems to allow that, and I'd avoid blocking that from the CRs because maybe it has a use case that we don't know of today. I'd hold this change until the conversation above is solved with e2e tests (which we need to implement anyway).

Reply:
for (a) you need to implememt https://datatracker.ietf.org/doc/html/draft-raszuk-mi-bgp-00 so u can have something like 
`router bgp 100 instance bgp1` and `router bgp 100 instance bgp2` so we will have two bgp speakers bgp1 and bgp2 both using myASN 100 and we can have two different sets of peers.

as far as I know metallb bgp stack run only one process hence only one MyASN will/should be used by all peers can u pls confirm @rata

my questions is more of very generic and not tied up to FRR

Reply:
> Oh, interesting, thanks! I guess, though, that the current metalLB stack works with different myASN just fine, right? 
I will be surprised if Metallb can do it